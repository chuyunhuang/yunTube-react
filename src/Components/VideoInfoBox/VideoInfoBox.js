import React from 'react';
import './VideoInfoBox.scss';
import { Image, Button } from 'semantic-ui-react';

export class VideoInfoBox extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      collapsed: true,
    };


  }


  render() {
    let descriptionTextClass = "collapsed";
    let buttonTitle = "Show More";
    if (!this.state.collapsed) {
      descriptionTextClass = "expand";
      buttonTitle = "Show Less"
    }
    return (
      <div className="video-info-box">
        <Image className='channel-image' src='http://via.placeholder.com/48x48' circular />
        <div className="video-info">
          <div className="channel-name">Yun is amazing</div>
          <div className="video-publish-date">Wed, Dec 4th</div>
        </div>
        <Button color="youtube" className="subscribe">1 Subscribe</Button>
        <div className="video-description">
          <div className={descriptionTextClass}>
            <p>1</p>
            <p>2</p>
            <p>3</p>
          </div>
          <Button compact onClick={this.onToggleCollapseButton}>{buttonTitle}</Button>
        </div>
      </div>

    )
  }

  onToggleCollapseButton = () => {
    this.setState((preState) => {
      return {
        collapsed: !preState.collapsed
      }
    })
  }
}