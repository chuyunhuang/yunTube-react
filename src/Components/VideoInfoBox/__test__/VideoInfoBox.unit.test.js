import React from 'react';
import { shallow } from 'enzyme';
import { VideoInfoBox } from '../VideoInfoBox';

describe('VideoInfoBox', () => {
  test('render VideoInfoBox in default mode, text collapsed', () => {
    const wrapper = shallow(<VideoInfoBox />)
    expect(wrapper).toMatchSnapshot();
  });

  test('render VideoInfoBox, text expand', () => {
    const wrapper = shallow(<VideoInfoBox />)
    wrapper.setState({ collapsed: false })
    expect(wrapper).toMatchSnapshot();
  });

})