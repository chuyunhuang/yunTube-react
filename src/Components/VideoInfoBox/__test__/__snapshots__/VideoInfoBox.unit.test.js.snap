// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoInfoBox render VideoInfoBox 1`] = `
<div
  className="video-info-box"
>
  <Image
    as="img"
    circular={true}
    className="channel-image"
    src="http://via.placeholder.com/48x48"
    ui={true}
  />
  <div
    className="video-info"
  >
    <div
      className="channel-name"
    >
      Yun is amazing
    </div>
    <div
      className="video-publish-date"
    >
      Wed, Dec 4th
    </div>
  </div>
  <Button
    as="button"
    className="subscribe"
    color="youtube"
  >
    1 Subscribe
  </Button>
  <div
    className="video-description"
  >
    <div
      className="collapsed"
    >
      <p>
        1
      </p>
      <p>
        2
      </p>
      <p>
        3
      </p>
    </div>
    <Button
      as="button"
      compact={true}
      onClick={[Function]}
    >
      Show More
    </Button>
  </div>
</div>
`;

exports[`VideoInfoBox render VideoInfoBox in default mode, text collapsed 1`] = `
<div
  className="video-info-box"
>
  <Image
    as="img"
    circular={true}
    className="channel-image"
    src="http://via.placeholder.com/48x48"
    ui={true}
  />
  <div
    className="video-info"
  >
    <div
      className="channel-name"
    >
      Yun is amazing
    </div>
    <div
      className="video-publish-date"
    >
      Wed, Dec 4th
    </div>
  </div>
  <Button
    as="button"
    className="subscribe"
    color="youtube"
  >
    1 Subscribe
  </Button>
  <div
    className="video-description"
  >
    <div
      className="collapsed"
    >
      <p>
        1
      </p>
      <p>
        2
      </p>
      <p>
        3
      </p>
    </div>
    <Button
      as="button"
      compact={true}
      onClick={[Function]}
    >
      Show More
    </Button>
  </div>
</div>
`;

exports[`VideoInfoBox render VideoInfoBox, text expand 1`] = `
<div
  className="video-info-box"
>
  <Image
    as="img"
    circular={true}
    className="channel-image"
    src="http://via.placeholder.com/48x48"
    ui={true}
  />
  <div
    className="video-info"
  >
    <div
      className="channel-name"
    >
      Yun is amazing
    </div>
    <div
      className="video-publish-date"
    >
      Wed, Dec 4th
    </div>
  </div>
  <Button
    as="button"
    className="subscribe"
    color="youtube"
  >
    1 Subscribe
  </Button>
  <div
    className="video-description"
  >
    <div
      className="expand"
    >
      <p>
        1
      </p>
      <p>
        2
      </p>
      <p>
        3
      </p>
    </div>
    <Button
      as="button"
      compact={true}
      onClick={[Function]}
    >
      Show Less
    </Button>
  </div>
</div>
`;
